### PR TITLE
Replace hostname -f with uname -n

### PR DESCRIPTION
**Pull request by nak3**
Should not depend on `hostname` command. The specification of `hostname` command is changeable. 
On the other hand, `uname -n` is a front end for a system call and simply get hostname. Please also refer gethostname(2) man page.

### CODE REVIEW

**Comment by nak3** [comment]
I signed it!

**Comment by dchen1107** [comment]
We use `hostname -f` because it returns fully qualified domain name (FQDN). If we changed it to `uname -n`, shouldn't we lost that?

**Comment by thockin-cc** [comment]
Didnt Yu-Ju just change it to only take the first word anyway?

**Comment by dchen1107** [comment]
@thockin If I remembered correctly Yu-Ju's change is involved only on static pods name. When generate fully name for those static pods, we only use the first word of current FQDN name. But I thought we first introduce fqdn because some cloud provider(s) expect hostname resolvable. Not sure if that is still the requirement now? 

@brendandburns

**Comment by brendandburns** [comment]
I don't think that we are wedded to FQDN, but it is very important that both the kubelet and the master have the same notion of the identity of a node.  kubelet currently gets this identity via ```hostname -f``` master generally gets it from a cloud provider.

As long as those match up, I don't think we much care.

**Comment by bgrant0607** [comment]
FWIW, there was a discussion a long time ago in #584.

**Comment by dchen1107** [comment]
Actually appending fqdn suffix already causes issue for hosted master on GKE based on @cjcullen today:

Today when NodeController "registers" nodes in SyncCloud by getting bare instance names (https://github.com/GoogleCloudPlatform/kubernetes/blob/9949936b5ce9d205328269ea304db4ab5dce5a67/pkg/cloudprovider/gce/gce.go#L101) from GCE, and recreating the fqdn and append it to instance names(https://github.com/GoogleCloudPlatform/kubernetes/blob/9949936b5ce9d205328269ea304db4ab5dce5a67/pkg/cloudprovider/gce/gce.go#L427). The issue is when the master is in a project we own, the fqdn suffix will be different, and we don't really have a good way of generating it on the master.

In this case, replacing hostname -f with uname -n is a right approach? 

@nak3 could you please run hack/e2e-test.sh?

File: pkg/cloudprovider/gce/gce.go [hunk 1]
```diff
@@ -394,10 +394,7 @@ func fqdnSuffix() (string, error) {
 	if err != nil {
 		return "", err
 	}
-	hostname, err := exec.Command("hostname").Output()
-	if err != nil {
-		return "", err
-	}
+	hostname := strings.Split(string(fullHostname), ".")[0]
```
Comment (dchen1107) [inline pkg/cloudprovider/gce/gce.go]: If we decide to not require fqdn suffix, this change is not required at all. Even more, we should remove the entire logic related to fqdn suffix here.

**Comment by zmerlynn** [comment]
uname -n doesn't seem to be deterministic, though? I have at least one box where it's giving me the FQDN and another it's giving me a naked name. It looks consistent with `hostname` (minus the -f), but if we need things to line up, that may mean we still have to name hack.

**Comment by zmerlynn** [comment]
Basically, I just repeated what @thockin said months ago: https://github.com/GoogleCloudPlatform/kubernetes/pull/584#issuecomment-49901915

**Comment by dchen1107** [comment]
hostname pretty much includes an internal call to uname, that explains why the outputs are the same. 
The problem is that hostname could be fully qualified or not qualified. POSIX does not specify which of these forms should be returned by uname. Some implementations do, but they are split between use of the unqualified name and the fully-qualified name.

My concern raised earlier is if our eco-system does require fully qualified name, and from various sources it sounds like NO. In this case, we could settle down with first token by splitting the output by "." to get rid of domain name. 

hostname -f failed us many times before because couldn't retrieve fqdn suffix.

**Comment by nak3** [comment]
@dchen1107
I will run hack/e2e-test.sh first. But hold on... I cannot pass hack/e2e-test.sh regardless of `hostname -f` or `uname -n`.

**Comment by dchen1107** [comment]
@nak3 What kind of failures are you observing? @quinton-hoole

**Comment by nak3** [comment]
@dchen1107

I am hiting similar error to #6045

I got same error with #6045, and after I updated, I still keep getting similar error.

* Commit ID
~~~
(master)$ git show -s --format=%H
2719194154ffd38fd1613699a9dd10a00909957e
~~~

* Log snippet

~~~
I0408 07:52:46.882759     728 integration.go:284] Error on creating endpoints: endpoints "service1" not found^M
I0408 07:52:47.482741     728 integration.go:284] Error on creating endpoints: endpoints "service1" not found^M
I0408 07:52:47.485097     728 integration.go:266] Check whether pod nginx-controller-h7l1o.test exists on node "localhost"^M

... snip ...

I0408 07:53:16.485701     728 integration.go:266] Check whether pod nginx-controller-moyc2.test exists on node ""^M
I0408 07:53:16.485712     728 integration.go:268] Pod nginx-controller-moyc2.test is not bound to a host yet^M
F0408 07:53:16.485718     728 integration.go:435] FAILED: pods never started running timed out waiting for the condition^M
!!! Error in hack/test-integration.sh:47^M
  '"${KUBE_OUTPUT_HOSTBIN}/integration" --v=2 --apiVersion="$1"' exited with status 255^M
Call stack:^M
  1: hack/test-integration.sh:47 runTests(...)^M
  2: hack/test-integration.sh:60 main(...)^M
Exiting with status 1^M
~~~

**Comment by dchen1107** [comment]
@nak3  Sorry for those failures tests, lately we have a churn of broken tests, and ppl are working on fix. I noticed that travis and shippable, which run integration test too, are green for this PR now. Can you sync your branch, and rerun hack/e2e-test.sh again? There are tons of output related pods, and events at the end of test output. Ignore those output and scroll up, you should see how many tests are passed, failed or skipped. There is one known failure. Please post the result here. Thanks!

**Comment by nak3** [comment]
@dchen1107

I synced and rerun hack/e2e-test.sh again. But I still got following error.

* Commit ID

~~~
(master)$  git show -s --format=%H
d85fbd7b5a402eea4959db71ad4afb87f4b7ae89
~~~


* Log
~~~
... snip ...
I0409 12:50:50.901542    1252 publish.go:152] setting endpoints for master service "kubernetes-ro" to &{{ } {kubernetes-ro  default    0001-01-01 00:00:00 +0000 UTC <nil> map[] map[]} [{[{172.17.0.50 <nil>}] [{ 7080 TCP}]}]}
E0409 12:50:52.341825    1252 publish.go:45] Can't create rw endpoints: endpoints "kubernetes" cannot be updated: the resource was updated to 88
E0409 12:50:52.343126    1252 publish.go:70] Can't create ro endpoints: endpoints "kubernetes-ro" cannot be updated: the resource was updated to 87
W0409 12:51:00.785274    1252 master.go:194] Portal net unspecified. Defaulting to 10.0.0.0/24.
I0409 12:51:00.785797    1252 master.go:224] Will report 172.17.0.50 as public IP address.
E0409 12:51:03.517143    1252 publish.go:42] Can't create rw service: Service "kubernetes" is invalid: spec.portalIP: invalid value '10.0.0.2': IP 10.0.0.2 is already allocated
I0409 12:51:03.525897    1252 publish.go:152] setting endpoints for master service "kubernetes" to &{{ } {kubernetes  default    0001-01-01 00:00:00 +0000 UTC <nil> map[] map[]} [{[{172.17.0.50 <nil>}] [{ 6443 TCP}]}]}
E0409 12:51:03.527285    1252 publish.go:67] Can't create ro service: Service "kubernetes-ro" is invalid: spec.portalIP: invalid value '10.0.0.1': IP 10.0.0.1 is already allocated
I0409 12:51:03.532114    1252 publish.go:152] setting endpoints for master service "kubernetes-ro" to &{{ } {kubernetes-ro  default    0001-01-01 00:00:00 +0000 UTC <nil> map[] map[]} [{[{172.17.0.50 <nil>}] [{ 7080 TCP}]}]}
E0409 12:51:04.045273    1252 publish.go:42] Can't create rw service: Service "kubernetes" is invalid: spec.portalIP: invalid value '10.0.0.2': IP 10.0.0.2 is already allocated
E0409 12:51:04.047512    1252 publish.go:67] Can't create ro service: Service "kubernetes-ro" is invalid: spec.portalIP: invalid value '10.0.0.1': IP 10.0.0.1 is already allocated
I0409 12:51:04.048992    1252 publish.go:152] setting endpoints for master service "kubernetes" to &{{ } {kubernetes  default    0001-01-01 00:00:00 +0000 UTC <nil> map[] map[]} [{[{172.17.0.50 <nil>}] [{ 6443 TCP}]}]}
I0409 12:51:04.050553    1252 publish.go:152] setting endpoints for master service "kubernetes-ro" to &{{ } {kubernetes-ro  default    0001-01-01 00:00:00 +0000 UTC <nil> map[] map[]} [{[{172.17.0.50 <nil>}] [{ 7080 TCP}]}]}
E0409 12:51:06.098020    1252 publish.go:70] Can't create ro endpoints: endpoints "kubernetes-ro" cannot be updated: the resource was updated to 102
E0409 12:51:06.099150    1252 publish.go:45] Can't create rw endpoints: endpoints "kubernetes" cannot be updated: the resource was updated to 101
E0409 12:51:13.181222    1252 publish.go:67] Can't create ro service: Service "kubernetes-ro" is invalid: spec.portalIP: invalid value '10.0.0.1': IP 10.0.0.1 is already allocated
E0409 12:51:13.186641    1252 publish.go:67] Can't create ro service: Service "kubernetes-ro" is invalid: spec.portalIP: invalid value '10.0.0.1': IP 10.0.0.1 is already allocated
I0409 12:51:13.187601    1252 publish.go:152] setting endpoints for master service "kubernetes-ro" to &{{ } {kubernetes-ro  default    0001-01-01 00:00:00 +0000 UTC <nil> map[] map[]} [{[{172.17.0.50 <nil>}] [{ 7080 TCP}]}]}
E0409 12:51:13.190653    1252 publish.go:42] Can't create rw service: Service "kubernetes" is invalid: spec.portalIP: invalid value '10.0.0.2': IP 10.0.0.2 is already allocated
E0409 12:51:13.193296    1252 publish.go:42] Can't create rw service: Service "kubernetes" is invalid: spec.portalIP: invalid value '10.0.0.2': IP 10.0.0.2 is already allocated
E0409 12:51:13.194463    1252 publish.go:67] Can't create ro service: Service "kubernetes-ro" is invalid: spec.portalIP: invalid value '10.0.0.1': IP 10.0.0.1 is already allocated
E0409 12:51:13.195583    1252 publish.go:42] Can't create rw service: Service "kubernetes" is invalid: spec.portalIP: invalid value '10.0.0.2': IP 10.0.0.2 is already allocated
I0409 12:51:13.196344    1252 publish.go:152] setting endpoints for master service "kubernetes-ro" to &{{ } {kubernetes-ro  default    0001-01-01 00:00:00 +0000 UTC <nil> map[] map[]} [{[{172.17.0.50 <nil>}] [{ 7080 TCP}]}]}
I0409 12:51:13.197442    1252 publish.go:152] setting endpoints for master service "kubernetes" to &{{ } {kubernetes  default    0001-01-01 00:00:00 +0000 UTC <nil> map[] map[]} [{[{172.17.0.50 <nil>}] [{ 6443 TCP}]}]}
I0409 12:51:13.198127    1252 publish.go:152] setting endpoints for master service "kubernetes" to &{{ } {kubernetes  default    0001-01-01 00:00:00 +0000 UTC <nil> map[] map[]} [{[{172.17.0.50 <nil>}] [{ 6443 TCP}]}]}
I0409 12:51:13.199839    1252 publish.go:152] setting endpoints for master service "kubernetes-ro" to &{{ } {kubernetes-ro  default    0001-01-01 00:00:00 +0000 UTC <nil> map[] map[]} [{[{172.17.0.50 <nil>}] [{ 7080 TCP}]}]}
I0409 12:51:13.203832    1252 publish.go:152] setting endpoints for master service "kubernetes" to &{{ } {kubernetes  default    0001-01-01 00:00:00 +0000 UTC <nil> map[] map[]} [{[{172.17.0.50 <nil>}] [{ 6443 TCP}]}]}
E0409 12:51:14.941478    1252 publish.go:67] Can't create ro service: Service "kubernetes-ro" is invalid: spec.portalIP: invalid value '10.0.0.1': IP 10.0.0.1 is already allocated
I0409 12:51:14.943950    1252 publish.go:152] setting endpoints for master service "kubernetes-ro" to &{{ } {kubernetes-ro  default    0001-01-01 00:00:00 +0000 UTC <nil> map[] map[]} [{[{172.17.0.50 <nil>}] [{ 7080 TCP}]}]}
E0409 12:51:14.946975    1252 publish.go:42] Can't create rw service: Service "kubernetes" is invalid: spec.portalIP: invalid value '10.0.0.2': IP 10.0.0.2 is already allocated
I0409 12:51:14.950540    1252 publish.go:152] setting endpoints for master service "kubernetes" to &{{ } {kubernetes  default    0001-01-01 00:00:00 +0000 UTC <nil> map[] map[]} [{[{172.17.0.50 <nil>}] [{ 6443 TCP}]}]}
E0409 12:51:14.956067    1252 publish.go:42] Can't create rw service: Service "kubernetes" is invalid: spec.portalIP: invalid value '10.0.0.2': IP 10.0.0.2 is already allocated
E0409 12:51:14.957719    1252 publish.go:67] Can't create ro service: Service "kubernetes-ro" is invalid: spec.portalIP: invalid value '10.0.0.1': IP 10.0.0.1 is already allocated
I0409 12:51:14.958566    1252 publish.go:152] setting endpoints for master service "kubernetes" to &{{ } {kubernetes  default    0001-01-01 00:00:00 +0000 UTC <nil> map[] map[]} [{[{172.17.0.50 <nil>}] [{ 6443 TCP}]}]}
I0409 12:51:14.959482    1252 publish.go:152] setting endpoints for master service "kubernetes-ro" to &{{ } {kubernetes-ro  default    0001-01-01 00:00:00 +0000 UTC <nil> map[] map[]} [{[{172.17.0.50 <nil>}] [{ 7080 TCP}]}]}
E0409 12:51:15.107128    1252 publish.go:70] Can't create ro endpoints: endpoints "kubernetes-ro" cannot be updated: the resource was updated to 112
E0409 12:51:15.114186    1252 publish.go:70] Can't create ro endpoints: endpoints "kubernetes-ro" cannot be updated: the resource was updated to 112
E0409 12:51:15.115267    1252 publish.go:45] Can't create rw endpoints: endpoints "kubernetes" cannot be updated: the resource was updated to 113
E0409 12:51:15.116286    1252 publish.go:45] Can't create rw endpoints: endpoints "kubernetes" cannot be updated: the resource was updated to 113
E0409 12:51:15.643778    1252 publish.go:45] Can't create rw endpoints: endpoints "kubernetes" cannot be updated: the resource was updated to 113
E0409 12:51:15.645251    1252 publish.go:70] Can't create ro endpoints: endpoints "kubernetes-ro" cannot be updated: the resource was updated to 112
E0409 12:51:15.646459    1252 publish.go:70] Can't create ro endpoints: endpoints "kubernetes-ro" cannot be updated: the resource was updated to 112
E0409 12:51:15.648037    1252 publish.go:45] Can't create rw endpoints: endpoints "kubernetes" cannot be updated: the resource was updated to 113
FAIL
FAIL	github.com/GoogleCloudPlatform/kubernetes/test/integration	113.699s
!!! Error in /go/src/github.com/GoogleCloudPlatform/kubernetes/hack/test-go.sh:152
  'go test "${goflags[@]:+${goflags[@]}}" ${KUBE_RACE} ${KUBE_TIMEOUT} "${@+${@/#/${KUBE_GO_PACKAGE}/}}"' exited with status 1
Call stack:
  1: /go/src/github.com/GoogleCloudPlatform/kubernetes/hack/test-go.sh:152 runTests(...)
  2: /go/src/github.com/GoogleCloudPlatform/kubernetes/hack/test-go.sh:207 main(...)
Exiting with status 1
!!! Error in hack/test-integration.sh:40
  'KUBE_GOFLAGS="-tags 'integration no-docker' " KUBE_RACE="-race" KUBE_TEST_API_VERSIONS="$1" "${KUBE_ROOT}/hack/test-go.sh" test/integration' exited with status 1
Call stack:
  1: hack/test-integration.sh:40 runTests(...)
  2: hack/test-integration.sh:60 main(...)
Exiting with status 1
+++ [0409 12:51:22] Integration test cleanup complete
!!! Error in /ssd/kubernetes/hack/e2e-internal/../../cluster/gce/../../cluster/../build/../build/common.sh:432
  '"${docker_cmd[@]}" "$@"' exited with status 1
Call stack:
  1: /ssd/kubernetes/hack/e2e-internal/../../cluster/gce/../../cluster/../build/../build/common.sh:432 kube::build::run_build_command(...)
  2: /ssd/kubernetes/hack/e2e-internal/../../cluster/gce/../../cluster/../build/release.sh:36 main(...)
Exiting with status 1
2015/04/09 21:51:26 Error running build-release: exit status 1
2015/04/09 21:51:26 Error building. Aborting.
exit status 1
~~~

**Comment by dchen1107** [comment]
This integration test failure is new. I suspect it is caused by #6466 itself. Have you tried to run hack/test-integration.sh in a different branch without your change? Looks like the resource version is messed up, but from your code, I couldn't link anything to that failure yet.

**Comment by dchen1107** [comment]
I suspect there is a race in our current HEAD. Will try to reproduce this.

**Comment by nak3** [comment]
@dchen1107

> Have you tried to run hack/test-integration.sh in a different branch without your change? 

Yes, I have tried to run without my change. 

> I suspect there is a race in our current HEAD. Will try to reproduce this.

Okay,  I will try to reproduce it tomorrow. (Sorry, I live in different time zone…)

File: docs/salt.md [hunk 2]
```diff
@@ -54,7 +54,7 @@ Key | Value
 `cbr-cidr` | (Optional) The minion IP address range used for the docker container bridge.
 `cloud` | (Optional) Which IaaS platform is used to host kubernetes, *gce*, *azure*, *aws*, *vagrant*
 `etcd_servers` | (Optional) Comma-delimited list of IP addresses the kube-apiserver and kubelet use to reach etcd.  Uses the IP of the first machine in the kubernetes_master role.
-`hostnamef` | (Optional) The full host name of the machine, i.e. hostname -f
+`hostnamef` | (Optional) The full host name of the machine, i.e. uname -n
```
Comment (cjcullen) [inline docs/salt.md]: It might be more descriptive to say "The network node hostname of the machine" if it is no longer going to be an FQDN.

**Comment by nak3** [comment]
@dchen1107

I cleaned up and sync latest commit, but I am still hitting another error.

* Commit ID

~~~
$ git show -s --format=%H
4f3f19511fd2efa29635fd8ff59bd9ec02456c22
~~~

* Log

~~~
... snip ...
+++ [0410 16:42:51] Docker builds done
+++ [0410 16:43:01] Building tarball: full
+++ [0410 16:43:01] Building tarball: test
2015/04/10 16:43:20 Running: get status
Project: myProject
Zone: us-central1-b
current-context: ""
Running: /var/tmp/fab/kubernetes/cluster/../cluster/gce/../../cluster/../_output/dockerized/bin/linux/amd64/kubectl --match-server-version version
Error: couldn't read version from server: Get http://localhost:8080/version: dial tcp 127.0.0.1:8080: connection refused

2015/04/10 16:43:21 Error running get status: exit status 1
2015/04/10 16:43:21 Running: up
Project: myProject
Zone: us-central1-b
Starting cluster using provider: gce
... calling verify-prereqs

All components are up to date.

All components are up to date.
... calling kube-up
Project: myProject
Zone: us-central1-b
current-context: ""
Running: /var/tmp/fab/kubernetes/hack/e2e-internal/../../cluster/../cluster/gce/../../cluster/../cluster/../cluster/gce/../../cluster/../_output/dockerized/bin/linux/amd64/kubectl config view -o template --template={{$dot := .}}{{with $ctx := index $dot "current-context"}}{{range $element := (index $dot "contexts")}}{{ if eq .name $ctx }}{{ with $user := .context.user }}{{range $element := (index $dot "users")}}{{ if eq .name $user }}{{ index . "user" "username" }}{{end}}{{end}}{{end}}{{end}}{{end}}{{end}}
current-context: ""
Running: /var/tmp/fab/kubernetes/hack/e2e-internal/../../cluster/../cluster/gce/../../cluster/../cluster/../cluster/gce/../../cluster/../_output/dockerized/bin/linux/amd64/kubectl config view -o template --template={{$dot := .}}{{with $ctx := index $dot "current-context"}}{{range $element := (index $dot "contexts")}}{{ if eq .name $ctx }}{{ with $user := .context.user }}{{range $element := (index $dot "users")}}{{ if eq .name $user }}{{ index . "user" "password" }}{{end}}{{end}}{{end}}{{end}}{{end}}{{end}}
Creating gs://kubernetes-staging-4484187ceb
Creating gs://kubernetes-staging-4484187ceb/...
BadRequestException: 400 Invalid argument.
2015/04/10 16:43:34 Error running up: exit status 1
2015/04/10 16:43:34 Error starting e2e cluster. Aborting.
exit status 1
~~~

**Comment by nak3** [comment]
@dchen1107
I cannot pass the test because of the above error. It is 100% reproduceable and even though I tried after new `git clone`. Do you pass the test without error? And should I submit as new issue?

**Comment by dchen1107** [comment]
@nak3 My tests are passed. It is flaky sometimes, but failed most of time. How about do a "make clean && make release && hack/e2e-test.sh"?

**Comment by cjcullen** [comment]
This currently breaks the /validate endpoint. It expects node names to be resolvable DNS addresses. That is a bug in the validate handler, but should be fixed before we merge this.

Also, is there a reason not to just use os.Hostname() in the Go code?

**Comment by nak3** [comment]
@dchen1107

I tried `make clean && make release && hack/e2e-test.sh` but it is still same result. Okay, I will try to debug it by myself.

@cjcullen

> Also, is there a reason not to just use os.Hostname() in the Go code?

Please read comment here:
https://github.com/GoogleCloudPlatform/kubernetes/blob/v0.15.0/pkg/util/node.go#L29-L30

os.Hostname() just returns from `/proc/sys/kernel/hostname`:
https://github.com/golang/go/blob/weekly.2012-03-27/src/pkg/os/sys_linux.go#L9-L26

**Comment by nak3** [comment]
Just a memo,

I resolved last issue, but I am still hitting another issue.

~~~
Found 2 nodes.
     1	e2e-test-knakayam-minion-g9cv.c.nak3-test2.internal
     2	e2e-test-knakayam-minion-lfas.c.nak3-test2.internal
Attempt 1 at checking Kubelet installation on node e2e-test-knakayam-minion-g9cv ... [not working yet]
Attempt 2 at checking Kubelet installation on node e2e-test-knakayam-minion-g9cv ... [not working yet]
Attempt 3 at checking Kubelet installation on node e2e-test-knakayam-minion-g9cv ... [not working yet]
Attempt 4 at checking Kubelet installation on node e2e-test-knakayam-minion-g9cv ... [not working yet]
Attempt 5 at checking Kubelet installation on node e2e-test-knakayam-minion-g9cv ... [not working yet]
Attempt 6 at checking Kubelet installation on node e2e-test-knakayam-minion-g9cv ... [not working yet]
Attempt 7 at checking Kubelet installation on node e2e-test-knakayam-minion-g9cv ...
Kubelet failed to install on node e2e-test-knakayam-minion-g9cv. Your cluster is unlikely to work correctly.
Please run ./cluster/kube-down.sh and re-create the cluster. (sorry!)
2015/04/17 00:25:14 Error running up: exit status 1
2015/04/17 00:25:14 Error starting e2e cluster. Aborting.
exit status 1
~~~

`$ ./cluster/kube-up.sh` didn't in my environment, and single running of `$ ./cluster/kube-up.sh` is hitting same problem.

**Comment by dchen1107** [comment]
@nak3 I believe the last issue you hit here is the issue mentioned by @cjcullen above: /validate endpoint is broken. I will take a deep look at this.

**Comment by dchen1107** [comment]
@nak3 Do you mind me to cherry-pick your commit to my branch, and pick up the rest of work from here? I will preserve your work, and fix what is broken internally.

**Comment by nak3** [comment]
@dchen1107

> Do you mind me to cherry-pick your commit to my branch, and pick up the rest of work from here? I will preserve your work, and fix what is broken internally.

No I don't mind. Please do as you like.

**Comment by dchen1107** [comment]
@nak3 I were out last Friday without sending my PR. It turns out several breakages in e2e setup and test itself were fixed recently. I sent out #7072 based on your commit, now everything works fine. Thanks!

**Comment by justinsb** [comment]
So this broke AWS.... but ... does this mean that the actual string doesn't matter any more (it doesn't need to be resolvable?)  If so, I might try putting in the AWS instance-id - using a truly unique identifier would make things a lot more robust.

**Comment by dchen1107** [comment]
@justinsb name is not required to be resolvable, but need to be unique in a given cluster. Using AWS instance-id is a good choice. Sorry for breakage here. I should cc to you at the first place. Thanks!

**Pull request closed without merging**